ClangPowerTools/ClangPowerTools/Commands/CommandController.cs

```diff
@@ -308,6 +314,29 @@ public void OnAfterRunCommand(object sender, CloseDataStreamingEventArgs e)
       {
         DisplayFinishedMessage(false);
       }
+
+      HasEncodingErrorEvent.Invoke(this, new EventArgs());
+    }
+
+    public void OnEncodingErrorDetected(object sender, HasEncodingErrorEventArgs e)
```

Comment:
Think about SOLID and separation of concerns principle for this method.

Reply:
Ok, I moved the method used for selecting items in ItemsCollector